File: src/main/java/commands/slashadapters/adapters/AIAdapterAbstract.java
package commands.slashadapters.adapters;

import commands.slashadapters.SlashAdapter;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;

public abstract class AIAdapterAbstract extends SlashAdapter {

    @Override
    public CommandData addOptions(CommandData commandData) {
        return commandData
                .addOption(OptionType.STRING, "image_url", "A link to the image", true);
    }

}

File: src/main/java/commands/slashadapters/adapters/AIAdapterAbstract.kt
package commands.slashadapters.adapters

import commands.slashadapters.SlashAdapter
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

abstract class AIAdapterAbstract : SlashAdapter() {

    public override fun addOptions(commandData: SlashCommandData): SlashCommandData {
        return commandData
            .addOptions(
                generateOptionData(OptionType.STRING, "image_url", "ai_imageurl", false),
                generateOptionData(OptionType.ATTACHMENT, "image_file", "ai_file", false)
            )
    }

}